[BUG] `DefaultAnthropicClient` SSE listener throws `MismatchedInputException` on non-object `data` (e.g. `[DONE]` sentinel)

<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->

**Describe the bug**
`DefaultAnthropicClient.createMessage(...)`'s `ServerSentEventListener.onEvent` deserializes the event payload **before** inspecting `event.event()`:

```java
// DefaultAnthropicClient.java:318
AnthropicStreamingData data = fromJson(event.data(), AnthropicStreamingData.class);

if ("message_start".equals(event.event())) { ... }
else if ("content_block_start".equals(event.event())) { ... }
...
```

If any SSE frame carries a `data:` line that is not a JSON object mappable to `AnthropicStreamingData`, `fromJson` throws:

```
com.fasterxml.jackson.databind.exc.MismatchedInputException:
Cannot deserialize value of type `...AnthropicStreamingData` from Array value (token `JsonToken.START_ARRAY`)
	at ...client.Json.fromJson(Json.java:29)
	at ...client.DefaultAnthropicClient$1.onEvent(DefaultAnthropicClient.java:318)
```

This happens for the terminal `data: [DONE]` sentinel emitted by OpenAI-compatible gateways/proxies in front of Claude. `DefaultServerSentEventParser` forwards every `data:` frame unfiltered (unlike the OpenAI parser, which drops `[DONE]`), so the Anthropic client receives `[DONE]` and crashes. The exception is swallowed by `ServerSentEventListenerUtils.ignoringExceptions`, but it pollutes logs and aborts trailing event processing.

**Log and Stack trace**
```
!ENTRY org.sterl.llmpeon 1 0 2026-06-08 15:46:40.686
!MESSAGE ServerSentEvent { event = "message_delta", data = "{"type":"message_delta","delta":{"stop_reason":"max_tokens","stop_sequence":null,"stop_details":null},"usage":{"input_tokens":70137,"cache_creation_input_tokens":0,"cache_read_input_tokens":0,"output_tokens":1024,"output_tokens_details":{"thinking_tokens":0}}     }" }

!ENTRY org.sterl.llmpeon 1 0 2026-06-08 15:46:40.686
!MESSAGE ServerSentEvent { event = "message_stop", data = "{"type":"message_stop"        }" }

!ENTRY org.sterl.llmpeon 1 0 2026-06-08 15:46:40.686
!MESSAGE ServerSentEvent { event = null, data = "[DONE]" }

!ENTRY org.sterl.llmpeon 1 0 2026-06-08 15:46:40.686
!MESSAGE Chatreponse: AiMessage { text = null, thinking = null, toolExecutionRequests = [], attributes = {} }

!ENTRY org.sterl.llmpeon 2 0 2026-06-08 15:46:40.686
!MESSAGE An exception occurred during the invocation of the SSE listener. This exception has been ignored.
!STACK 0
java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `dev.langchain4j.model.anthropic.internal.api.AnthropicStreamingData` from Array value (token `JsonToken.START_ARRAY`)
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1]
at dev.langchain4j.model.anthropic.internal.client.Json.fromJson(Json.java:29)
at dev.langchain4j.model.anthropic.internal.client.DefaultAnthropicClient$1.onEvent(DefaultAnthropicClient.java:318)
at dev.langchain4j.http.client.log.LoggingHttpClient$1.onEvent(LoggingHttpClient.java:88)
at dev.langchain4j.http.client.sse.DefaultServerSentEventParser.lambda$parse$0(DefaultServerSentEventParser.java:28)
at dev.langchain4j.http.client.sse.ServerSentEventListenerUtils.ignoringExceptions(ServerSentEventListenerUtils.java:14)
at dev.langchain4j.http.client.sse.DefaultServerSentEventParser.parse(DefaultServerSentEventParser.java:28)
at dev.langchain4j.http.client.jdk.JdkHttpClient.lambda$execute$0(JdkHttpClient.java:84)
at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:739)
at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:531)
at java.base/java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:635)
at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:865)
at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:504)
at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450)
at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019)
at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `dev.langchain4j.model.anthropic.internal.api.AnthropicStreamingData` from Array value (token `JsonToken.START_ARRAY`)
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1]
at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:72)
at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1814)
at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1588)
at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1535)
at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeFromArray(BeanDeserializer.java:687)
at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:203)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:179)
at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:5052)
at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3954)
at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3922)
at dev.langchain4j.model.anthropic.internal.client.Json.fromJson(Json.java:27)
... 15 more
```

**To Reproduce**
Stream from any Claude-compatible endpoint that emits a trailing `data: [DONE]` frame (common with reverse proxies / OpenAI-shim gateways). Observed payloads:

```
event = "message_delta", data = {"type":"message_delta","delta":{"stop_reason":"max_tokens",...}}
event = "message_stop",  data = {"type":"message_stop"}
event = null,            data = "[DONE]"     <-- crash here
```

**Expected behavior**
Error is thrown ->  {"type":"message_delta","delta":{"stop_reason":"max_tokens",...}}

Maybe
Unknown / non-object event payloads (especially the `[DONE]` sentinel and frames with `event == null`) should be ignored gracefully — no parse attempt, no exception.

**Please complete the following information:**
- LangChain4j version: 1.16.1
- LLM(s) used: Opus
- Java version: 21
- Spring Boot version (if applicable): -

**Suggestion**
Guard before deserializing: skip frames where `event.event() == null`, where `data` equals `[DONE]`, or where `data` does not start with `{`. Only deserialize known Anthropic event types.

--- a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/client/DefaultAnthropicClient.java
+++ b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/client/DefaultAnthropicClient.java
@@ -315,25 +315,53 @@ public class DefaultAnthropicClient extends AnthropicClient {
                     streamingHandle = toStreamingHandle(context.parsingHandle());
                 }
 
-                AnthropicStreamingData data = fromJson(event.data(), AnthropicStreamingData.class);
+                String eventName = event.event();
+                String eventData = event.data();
 
-                if ("message_start".equals(event.event())) {
+                // OpenAI-compatible gateways in front of Claude may emit a trailing
+                // "data: [DONE]" sentinel or frames with an unknown/missing event name.
+                // Skip them gracefully instead of attempting to deserialize as
+                // AnthropicStreamingData, which would throw MismatchedInputException.
+                if (isSkippableSseFrame(eventName, eventData)) {
+                    rawServerSentEvents.add(event);
+                    return;
+                }
+
+                AnthropicStreamingData data = fromJson(eventData, AnthropicStreamingData.class);
+
+                if ("message_start".equals(eventName)) {
                     handleMessageStart(data);
-                } else if ("content_block_start".equals(event.event())) {
+                } else if ("content_block_start".equals(eventName)) {
                     handleContentBlockStart(data, streamingHandle);
-                } else if ("content_block_delta".equals(event.event())) {
+                } else if ("content_block_delta".equals(eventName)) {
                     handleContentBlockDelta(data, streamingHandle);
-                } else if ("content_block_stop".equals(event.event())) {
+                } else if ("content_block_stop".equals(eventName)) {
                     handleContentBlockStop(data, streamingHandle);
-                } else if ("message_delta".equals(event.event())) {
+                } else if ("message_delta".equals(eventName)) {
                     handleMessageDelta(data);
-                } else if ("message_stop".equals(event.event())) {
+                } else if ("message_stop".equals(eventName)) {
                     handleMessageStop();
-                } else if ("error".equals(event.event())) {
+                } else if ("error".equals(eventName)) {
                     handleError(data);
                 }
 
                 rawServerSentEvents.add(event);
+            }
+
+            private static boolean isSkippableSseFrame(String eventName, String eventData) {
+                if (eventName == null) {
+                    return true;
+                }
+                if (eventData == null) {
+                    return true;
+                }
+                String trimmed = eventData.trim();
+                if (trimmed.isEmpty() || "[DONE]".equals(trimmed)) {
+                    return true;
+                }
+                // Anthropic SSE payloads are JSON objects; anything else (arrays, scalars)
+                // cannot be deserialized into AnthropicStreamingData.
+                return !trimmed.startsWith("{");
             }
 
             private void handleMessageStart(AnthropicStreamingData data) {

--- a/langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/internal/client/DefaultAnthropicClientTest.java
+++ b/langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/internal/client/DefaultAnthropicClientTest.java
@@ -414,6 +414,55 @@ class DefaultAnthropicClientTest {
         }
 
         @Test
+        void shouldIgnoreDoneSentinelAndUnknownEventFrames() throws Exception {
+            // Given: a normal stream followed by a trailing "data: [DONE]" sentinel
+            // (emitted by OpenAI-compatible proxies in front of Claude) and a frame
+            // with no event name. Both must be skipped without throwing.
+            List<ServerSentEvent> events = List.of(
+                    createMessageStartEvent(),
+                    createContentBlockStartEvent(),
+                    createContentBlockDeltaEvent(),
+                    createContentBlockStopEvent(),
+                    createMessageDeltaEvent(),
+                    createMessageStopEvent(),
+                    new ServerSentEvent(null, "[DONE]"),
+                    new ServerSentEvent("ping", "{}"));
+            MockHttpClient mockHttpClient = MockHttpClient.thatAlwaysResponds(events);
+
+            DefaultAnthropicClient subject = createClient(mockHttpClient);
+
+            AnthropicCreateMessageRequest request = createMessageRequest();
+
+            CompletableFuture<ChatResponse> futureResponse = new CompletableFuture<>();
+            StringBuilder partialResponses = new StringBuilder();
+
+            // When
+            subject.createMessage(request, new StreamingChatResponseHandler() {
+                @Override
+                public void onPartialResponse(String partialResponse) {
+                    partialResponses.append(partialResponse);
+                }
+
+                @Override
+                public void onCompleteResponse(ChatResponse completeResponse) {
+                    futureResponse.complete(completeResponse);
+                }
+
+                @Override
+                public void onError(Throwable error) {
+                    futureResponse.completeExceptionally(error);
+                }
+            });
+
+            ChatResponse response = futureResponse.get(5, TimeUnit.SECONDS);
+
+            // Then: the stream completes normally and the [DONE] / unknown frames do not
+            // corrupt the aggregated text or trigger an error.
+            assertThat(response).isNotNull();
+            assertThat(partialResponses.toString()).contains("Hello", " world");
+        }
+
+        @Test
         void shouldHandleInterleavedParallelToolCalls() throws Exception {
             // Given: SSE events where two tool calls are interleaved
             // (content_block_start for index=2 arrives before content_block_stop for index=1)

